modules/host/negotiaterevisecontract_test.go

```diff
@@ -0,0 +1,88 @@
+package host
+
+import (
+	"testing"
+)
+
+func TestShannonEntropy(t *testing.T) {
+	t.Parallel()
+	uniform := make([]byte, 256)
+	for i := 0; i < 256; i++ {
+		uniform[i] = byte(i)
+	}
+	almost := make([]byte, 256)
+	for i := 0; i < 256-10; i++ {
+		almost[i] = byte(i)
+	}
+	testCases := []struct {
+		name      string
+		input     []byte
+		low, high float64
+	}{
+		{
+			name:  "all zeros",
+			input: make([]byte, 100),
+			low:   0.0,
+			high:  0.00000001,
+		},
+		{
+			name:  "uniform distrinution",
+			input: uniform,
+			low:   0.9999999,
+			high:  1.0,
```

Comment:
it's a little surprising that 1,2,3... would yield such a high result, but I guess it makes sense under the definition of Shannon entropy. Still, makes me wonder if a compression test is warranted.

Reply:
gzip extends the length of {0, 1, 2, ..., 256} from 256 bytes to 284 bytes even with the best compression level: https://play.golang.org/p/uJSjjQTepx (Probably gzip's algorithm does not recognize this kind of pattern.)

I think we do not need to catch this sequence to fix #2260 because media files (which we hunt for) are unlikely to have this sequence.

Reply:
interesting, thanks for running the test. In that case, I think Shannon entropy is good enough for now.